Fail build when unittests fail for PRs

BertScholten opened the pull request:
Be more strict on failing tests for PRs, to avoid missing a failing test before merging.

Hilbrand reviewed (approved):
LGTM

BertScholten committed:
Include all relevant modules to publish profile

Not sure what the idea was behind not publishing the service and its dependencies, but it's causing the code not to be compiled/tested when building the PR. And these should be published to nexus anyway I think?

BertScholten committed:
re-introduce the PR action

it was removed with f17cdc8d3041797ed050e3856ca9e17ca38c66ef, which seemed a bit weird? Assume it was just supposed to remove just the docker build part.

BertScholten committed:
Adjust jenkinsfile for PRs to not skip tests

Also not specifying a profile, as search does not have a 'deploy' profile anyway.

SerhatG commented on .github/workflows/on-pull_request-opened-synchronize-reopened.yml:
```diff
@@ -0,0 +1,22 @@
+name: pull_request-opened-synchronize-reopened-event
```
I believe we talked about it in a Technical Meeting and decided to move it to Jenkins when it concerns an environment that will be build and deployed using Jenkins. (as to not do things twice)

Yeah, I'd just remove this.

BertScholten committed:
Remove PR action again

Jenkins already builds on PR, no real need to do it twice. Though jenkins isn't publicly reachable, so not as transparent for people outside AERIUS, it's also not like we had people outside contributing.
